embedded/drivers/README.md

```diff
@@ -30,3 +30,29 @@ int write_data(uint8_t, uint8_t val); // Write data to a register
 int read_data(uint8_t *buff, int size); //Read data from a device
 uint8_t read_from_reg(uint8_t reg); // Read data from a specific register
 ```
+
+## Analog class
+
+This class represents a single analog pin on the Beaglebone Black.
+
+### Getting started
+
+To create a new analog pin object, call the constructor `Analog::Analog`. This constructor takes the pin's number as well as a scale factor to multiply the measured voltage by.
+
+```c++
+Analog::Analog(int pin, float scale)
+```
+
+Then, call the `Analog::begin` function to initialize the resources required to read from the analog pin.
+
```

Comment:
```suggestion
Example:
```c++
Analog potentiometerIn = Analog(7, 1.0);
if (!potentiometerIn.begin()) {
    printf("Potentiometer in pin failed to begin!\n\r");
}
```
```

Reply:
added example